juju/juju
# [JUJU-3759] Remove cached model from the facade context

The model cache was just used to provide some of the data needed to populate the status result.
Most of the status result is read from state anyway; there's no need for the model cache anymore

## QA steps

deploy juju-qa-dummy-source
check status as machine is being provisioned
after unit is deployed, check app and unit status matches
status-set just the unit status and check it is updated but app status stays the same
status-set the app status and check that it is updated and unit status is unchanged

on k8s, check with snappass-test and juju-mariadb-k8s
